src/v/raft/state_machine_manager.cc

```diff
@@ -0,0 +1,453 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "raft/state_machine_manager.h"
+
+#include "bytes/iostream.h"
+#include "config/property.h"
+#include "model/fundamental.h"
+#include "model/timeout_clock.h"
+#include "raft/consensus.h"
+#include "raft/logger.h"
+#include "raft/state_machine_base.h"
+#include "raft/types.h"
+#include "ssx/future-util.h"
+#include "ssx/semaphore.h"
+#include "storage/types.h"
+#include "utils/mutex.h"
+
+#include <seastar/core/shared_ptr.hh>
+#include <seastar/core/with_scheduling_group.hh>
+#include <seastar/coroutine/maybe_yield.hh>
+#include <seastar/coroutine/parallel_for_each.hh>
+#include <seastar/coroutine/switch_to.hh>
+
+#include <stdexcept>
+#include <vector>
+
+namespace raft {
+
+/**
+ * Applicator which is a batch consumer that applies the same batch to multiple
+ * state machines. If one of the STMs throws an exception from `apply` method
+ * then not further batches are applied to that STM but the others continue
+ */
+class batch_applicator {
```

Comment:
batch applicator access private fields of `state_machine_manager` i need it to be a `friend` class of manager, if i put in anonymous ns it will be impossible, i can make it inner class of `state_machine_manager` WDYT ?

Reply:
Ah no this is good, I missed the friending.